pytypes/type_util.py

```diff
@@ -216,16 +216,17 @@ def get_orig_class(obj, default_to__class__=False):
             cls = obj.__class__
         if _typing_3_7 and is_Generic(cls):
             # Workaround for https://github.com/python/typing/issues/658
-            stck = stack()
+            frame = currentframe()
             # Searching from index 2 is sufficient: At 0 is get_orig_class, at 1 is the caller.
             # We assume the caller is not typing._GenericAlias.__call__ which we are after.
-            for line in stck[2:]:
+            while frame and frame.f_lineno >= 2:
```

Comment:
What does the check for `f_lineno` do here? It makes no sense to me.

Reply:
Probably, @bhavints thought it would yield the frame number. TBH I do not really know right now what it actually does. I guess the source code line? Do you happen to know whether the result of ``currentframe()`` is ``stack[0]`` or ``stack[-1]``? (I suppose the latter)